model/generate_test.go

```diff
@@ -1694,15 +1709,23 @@ func TestSimulateNewDependencyGraph(t *testing.T) {
 	})
 
 	t.Run("CreatesLoop", func(t *testing.T) {
+		// Generator generates: [ A (depends on B), B (depends on A) ]
+		// the graph:
+		//   +-----+ ---------> +-----+
+		//   |  A  |            |  B  |
+		//   +-----+ <--------- +-----+
+
 		project := &Project{
 			BuildVariants: []BuildVariant{
 				{Name: "bv0", Tasks: []BuildVariantTaskUnit{
-					{Name: "generated", Variant: "bv0", DependsOn: []TaskUnitDependency{{Name: "generator", Variant: "bv0"}}},
```

Comment:
Note: This setup is recreated as a non error case and explained more in the "Dependent on Generator ('loop')" test case.